packages/ui/src/components/Slider/components/SingleSlider.tsx

```diff
@@ -124,24 +120,21 @@ export const SingleSlider = ({
     return []
   }, [options])
 
-  const onChangeCallback = useCallback(
-    (newValue: number) => {
-      setValueToShow(options ? options[newValue].value : newValue)
-      setComputedValue(newValue ?? min)
-      onChange?.(newValue ?? min)
-    },
-    [min, onChange, options],
-  )
+  const internalOnChangeRef = useRef((newValue: number) => {
+    setSelectedIndex(newValue ?? min)
+    onChange?.(newValue ?? min)
+  })
 
   // Make sure that min <= value <= max
   useEffect(() => {
     if (value < min) {
```

Comment:
can we create a function for this one ? i think it might be used somewhere else ?

Reply:
We dont use this anywhere else, its just to ensure that the component is working when passing a bad value